Branch-SDK/Branch-SDK/BNCPreferenceHelper.m

```diff
@@ -609,38 +609,62 @@ - (void)writeObjectToDefaults:(NSString *)key value:(NSObject *)value {
     [self persistPrefsToDisk];
 }
 
-- (void)persistPrefsToDisk {
-    NSDictionary *persistenceDict = [self.persistenceDict copy];
-    NSBlockOperation *newPersistOp = [NSBlockOperation blockOperationWithBlock:^{
-        if (![NSKeyedArchiver archiveRootObject:persistenceDict toFile:[self prefsFile]]) {
-            [self logWarning:@"Failed to persist preferences to disk"];
+- (void)persistPrefsToDisk
+    {
+    @synchronized (self)
+        {
+        NSDictionary *persistenceDict = [self.persistenceDict copy];
+        NSBlockOperation *newPersistOp = [NSBlockOperation blockOperationWithBlock:
+        ^   {
+            NSData *data = [NSKeyedArchiver archivedDataWithRootObject:persistenceDict];
+            if (!data)
+                {
+                [self logWarning:@"Can't create preferences archive."];
+                return;
+                }
+            NSError *error = nil;
+            [data writeToURL:self.class.URLForPrefsFile
+                options:NSDataWritingAtomic error:&error];
+            if (error)
+                {
+                [self logWarning:
+                    [NSString stringWithFormat:
+                        @"Failed to persist preferences to disk: %@.", error]];
+                }
+            }];
+        [self.persistPrefsQueue addOperation:newPersistOp];
         }
-    }];
-    [self.persistPrefsQueue addOperation:newPersistOp];
-}
+    }
 
 #pragma mark - Reading From Persistence
 
-- (NSMutableDictionary *)persistenceDict {
-    if (!_persistenceDict) {
+- (NSMutableDictionary *)persistenceDict
+    {
+    if (!_persistenceDict)
+        {
         NSDictionary *persistenceDict = nil;
         @try {
-            persistenceDict = [NSKeyedUnarchiver unarchiveObjectWithFile:[self prefsFile]];
-        }
-        @catch (NSException *exception) {
-            [self logWarning:@"Failed to load preferences from disk"];
-        }
+            NSError *error = nil;
+            NSData *data =
+                [NSData dataWithContentsOfURL:self.class.URLForPrefsFile
```

Comment:
This conditional newline for instantiations is quite a deviation (and hurting my eyes. Can you put this on the same line as the variable?